api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -301,6 +301,46 @@ message HttpConnectionManager {
     type.http.v3.PathTransformation http_filter_transformation = 2;
   }
 
+  // Configures the manner in which the Proxy-Status HTTP response header is
+  // populated.
+  //
+  // See the Proxy-Status RFC: https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-proxy-status-05
+  // TODO(ambuc): Update this with the non-draft URL when finalized.
+  //
+  // The Proxy-Status header is a string of the form:
+  //
+  //   "<server_name>; error=<error_type>; details=<details>"
+  message ProxyStatusConfig {
+    enum ProxyName {
+      // Use the string "envoy".
+      ENVOY_LITERAL = 0;
+
+      // Use the name of the server as it appears in
+      // HttpConnectionManager.server_name.
+      SERVER_NAME = 1;
```

Comment:
do we have a use case for sending something other than server name?  Given server name is configurable already I'd lean towards nixing this config unless someone asks for it.

Reply:
I was thinking of the scenario where a deployment of Envoy would use something identifying (server location, etc.) within the Envoy's server_name, but the specific server_name shouldn't be exposed to whoever receives traffic from that Envoy. With a configurable proxy name, you could still use a specific server_name in configs, logs, etc. while obscuring that name from traffic. WDYT?

Reply:
Suggested in a different comment thread, but adding here too:
Another option, if you wish to make this more configurable, is to convert `proxy_name` to be of type `string` that defaults to server_name. This way if one wishes to obscure the name from traffic, they can explicitly set a desired name here.

Reply:
I have a use case that I don't think fits in either of these cases yet.

We set `server_name` to the name of our product so that the Server header has our product name in it.  However, for Proxy-Status, we would like to identify the particular instance of our server that generated the Proxy-Status line.  This is almost the opposite of what @ambuc suggests in their comment.